R/zchunk_L242.ssp34_pasture.R

```diff
@@ -0,0 +1,161 @@
+    A_LandLeaf_Unmgd2 <- get_data(all_data, "aglu/A_LandLeaf_Unmgd2")
+    A_LandLeaf2 <- get_data(all_data, "aglu/A_LandLeaf2")
+    get_data(all_data, "L125.LC_bm2_R_LT_Yh_GLU") %>%
+      replace_GLU(basin_to_country_mapping) ->
+      L125.LC_bm2_R_LT_Yh_GLU
+    L102.pcgdp_thous90USD_Scen_R_Y <- get_data(all_data, "L102.pcgdp_thous90USD_Scen_R_Y")
+
+    # Build tables
+
+    FRACT_UNMGD_TO_MGD <- 0.25
+
+    # LAND ALLOCATION AND LAND USE HISTORY
+    # Subset the relevant land types from table with land cover by all land types
+    # Unmanaged land (lines 43-58 in old file)
+    L125.LC_bm2_R_LT_Yh_GLU %>%
+      filter(Land_Type == "UnmanagedPasture") %>%
+      left_join_error_no_match(GCAM_region_names, by = "GCAM_region_ID") %>%
+      # Adjust land cover downwards so more pasture is in the managed category
+      mutate(allocation = value * (1 - FRACT_UNMGD_TO_MGD)) %>%
+      add_node_leaf_names(nesting_table = A_LandLeaf_Unmgd2, leaf_name = "UnmanagedLandLeaf", "LandNode1", "LandNode2") ->
+      L242.LC_bm2_R_Unmgd2_Yh_GLU.mlt
+
+    # Historical land cover, unmanaged land in the second nest
+    L242.LC_bm2_R_Unmgd2_Yh_GLU.mlt %>%
+      filter(year %in% LAND_HISTORY_YEARS) %>%
+      select(region, LandAllocatorRoot, LandNode1, LandNode2, UnmanagedLandLeaf, year, allocation) ->
+      L242.LN2_HistUnmgdAllocation_ALL
+
+    # Model base period land cover, unmanaged land in the second nest
+    L242.LC_bm2_R_Unmgd2_Yh_GLU.mlt %>%
+      filter(year %in% BASE_YEARS) %>%
+      select(region, LandAllocatorRoot, LandNode1, LandNode2, UnmanagedLandLeaf, year, allocation) ->
+      L242.LN2_UnmgdAllocation_ALL
+
+    # Managed land (60-75)
+    L125.LC_bm2_R_LT_Yh_GLU %>%
+      filter(Land_Type == "Pasture") %>%
+      left_join_error_no_match(GCAM_region_names, by = "GCAM_region_ID") %>%
+      left_join_error_no_match(select(L242.LC_bm2_R_Unmgd2_Yh_GLU.mlt, GCAM_region_ID, year, GLU, value), by = c("GCAM_region_ID", "year", "GLU")) %>%
+      mutate(allocation = value.x + FRACT_UNMGD_TO_MGD * value.y) %>%
+      select(GCAM_region_ID, Land_Type, GLU, year, value = value.x, region, allocation) %>%
+      add_node_leaf_names(nesting_table = A_LandLeaf2, leaf_name = "LandLeaf", "LandNode1", "LandNode2") ->
+      L242.LC_bm2_R_Mgd2_Yh_GLU.mlt
+
+    # Historical land cover, managed land in the second nest
+    L242.LC_bm2_R_Mgd2_Yh_GLU.mlt %>%
+      filter(year %in% LAND_HISTORY_YEARS) %>%
+      select(region, LandAllocatorRoot, LandNode1, LandNode2, LandLeaf, year, allocation) ->
+      L242.LN2_HistMgdAllocation_ALL
+
+    # Model base period land cover, managed land in the second nest
+    L242.LC_bm2_R_Mgd2_Yh_GLU.mlt %>%
+      filter(year %in% BASE_YEARS) %>%
+      select(region, LandAllocatorRoot, LandNode1, LandNode2, LandLeaf, year, allocation) ->
+      L242.LN2_MgdAllocation_ALL
+
+    # Create SSP4 pasture inputs, isolating low-growth regions for subsequent filtering
```

Comment:
The filter is for regions with very low GDP, looks like; call them "poor (low GDP) regions"?